modules/c++/sys/source/OSUnix.cpp

```diff
@@ -132,13 +133,68 @@ std::set<std::string> get_unique_thread_siblings()
     return unique_ts;
 }
 
-void get_cpu_affinity(cpu_set_t* mask)
+class ScopedCPUAffinity
 {
-    if (-1 == sched_getaffinity(0, sizeof(cpu_set_t), mask))
+public:
+    ScopedCPUAffinity()
     {
-        throw except::Exception(Ctxt("Failed to get CPU affinity"));
+        int numOnlineCPUs = sysconf(_SC_NPROCESSORS_ONLN);
+        if (numOnlineCPUs == -1)
+        {
+            throw except::Exception(Ctxt("Failed to get online CPU count"));
+        }
+
+        // The number of processors in the mask must be _at least_ the number of
+        // CPU's representable by the kernel's internal mask. This is given by
+        // the constant CPU_SETSIZE.
+        int maxCPUs = std::max(numOnlineCPUs, CPU_SETSIZE);
```

Comment:
Can be const.  Also, is CPU_SETSIZE just a `#define`?  If so, use `std::max<int>()` here just to make sure a compiler doesn't get confused and can't find a matching type or something dumb.